packages/compiler/src/server/types.ts

```diff
@@ -70,6 +77,17 @@ export interface CompileResult {
   readonly optionsFromConfig: CompilerOptions;
 }
 
+export interface ServerDiagnostic extends Diagnostic {
+  target: (SourceLocation & { position?: { line: number; column: number } }) | typeof NoTarget;
+}
+
+export interface CustomCompileResult {
```

Comment:
Better to call it InternalCompileResult to be consistent with the internalCompile()

Reply:
updated